ci-operator/step-registry/dpu-operator/e2e-tests-workflow/tests/dpu-operator-e2e-tests-workflow-tests-commands.sh

```diff
@@ -1,13 +1,24 @@
 #!/bin/bash
 
+check_job_status() {
+    job_check=$(curl -k -s --resolve "$endpoint_resolve" "$job_url/api/json")
+    result=$(echo "$job_check" | jq -r '.result')
+    
+    if [ "$result" != "null" ]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+
 token_dpu_operator_key=$(cat "/var/run/token/dpu-token/dpu-key")
 endpoint=$(cat "/var/run/token/dpu-token/url")
 
-job_url="https://${endpoint}/job/99_Lab217_IPU_Deploy"
-
+job_url="https://${endpoint}/job/99_Lab217_E2E_IPU_Deploy"
 endpoint_resolve="${endpoint}:443:10.0.180.88"
 
-max_sleep_duration=86400  # Maximum sleep duration in seconds (2 hours)
+max_sleep_duration=432000  # Maximum sleep duration in seconds (5 days)
```

Comment:
why did you bump this to 5 days?

Reply:
This is just a safeguard. In case there are a lot of PRs I don't want it to timeout prematurely.

Reply:
Multiple PRs can be in the middle of this script in parallel?

Reply:
No, what this code does is wait for other PRs to finish executing PRIOR to firing. Hence the need for the long timetout. Let's say there are like 20 PR's in the queue. It is going to take a long time to clear everything out.

Reply:
If multiple PRs **can not** be in the middle of this script in parallel then why do we need this timeout? If multiple instances of this script would be running in PROW then would there be a better way to make sure prow serializes them instead of running them in parallel.

Reply:
I don't think there is any way to stop prow from running multiple instances of this script in parallel. Did you have something in mind?

Reply:
https://github.com/openshift/release/blob/master/ci-operator/step-registry/telco5g/cnf/tests/telco5g-cnf-tests-commands.sh

Does the Telco 5g test also have such a wait in their scripts? It just seems quite fragile to wait and block on the output of the Jenkins job. It seems that we could potentially have a race.

Reply:
E.g. Reading from the Job (Result is empty) -> do stuff -> (Jenkins Job Result is now not empty due to another running PR) -> Try to run the Job.

The thing is that there is no proper mutual exclusion locking.